### PR TITLE
feat(appsync): add RDS datasource

### LINKED ISSUE #9152: [aws-appsync] Add Relational Database (rds) datasource
Currently the `GraphQLApi` only supports dynamo, HTTP, none, and lambda data sources and is missing the Relational Database source.

### Use Case

When setting up an RDS datasource currently can't use CDK to create that datasource.

### Proposed Solution

Create an RDS datasource. I'm happy to get moving on this if no one has started yet.

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by kochie**
~~Very, ***very*** preliminary attempt at adding RDS data source to AppSync.~~

~~Still need to fix tests and lint.~~

This PR adds support for RDS as a datasource for AppSync.

There are several examples included in the README, integration tests, and documentation.

Fixes #9152 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-appsync/lib/data-source.ts
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/lib/data-source.ts:283]: I believe you need a grant statement here to give `appsync` the ability to get/update/etc. the RDS cluster and to get the secret value for the secret. See the policy statement for appsync rds [here](https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-rds-resolvers.html#graphql-schema).

See the dynamodb data source implementation for reference:
https://github.com/aws/aws-cdk/blob/975efbed5518c4b4b982038ee6a4ad79fe11935d/packages/%40aws-cdk/aws-appsync/lib/data-source.ts#L196-L200

----

In this case, I believe the grants needed are as follows.

For secret:
```ts
props.secretStore.grantRead(this);
```

For rds:

There is no rds grant methods at the moment. Which means we would need a PR to add a grant method in rds to provide the equivalent of:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "rds-data:DeleteItems",
                "rds-data:ExecuteSql",
                "rds-data:ExecuteStatement",
                "rds-data:GetItems",
                "rds-data:InsertItems",
                "rds-data:UpdateItems"
            ],
            "Resource": [
                "arn:aws:rds:us-east-1:123456789012:cluster:mydbcluster",
                "arn:aws:rds:us-east-1:123456789012:cluster:mydbcluster:*"
            ]
        }
    ]
}
```

@skinny85 wdyt? I think its either adding a grant or to just make a Iam policy here which im not a fan of

File: packages/@aws-cdk/aws-appsync/README.md
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:106]: nit: i dont think the addition of rds data source is necessary

the README functions to help understand structure.. implementation of each should be conveyable through the docstrings

Comment (BryanPan342) [reply packages/@aws-cdk/aws-appsync/README.md:106]: instead something more usable would be a section that described the rds data source

```
### Aurora Severless

// Description of service
To add a Amazon Aurora Serverless cluster as a datasource, use the...

// Example of usage
```ts
usage..
.```

// Note key api needs
Note: A Secret is needed for ...
```
but i defer to @MrArnoldPalmer for final say on this

File: packages/@aws-cdk/aws-appsync/lib/data-source.ts
Comment (BryanPan342) [reply packages/@aws-cdk/aws-appsync/lib/data-source.ts:283]: spoke to @skinny85 offline and came to the conclusion the best way to do this would be to just make the policy here and put a note to change it once rds implements grants.

the implementation would look something like this
```ts
const clusterArn = cdk.Stack.of(this).formatArn({
    service: 'rds',
    resource: `cluster:${props.databaseCluster.clusterIdentifier}`,
});

iam.Grant.addToPrincipal({
  grantee: this,
  actions: [ 
     "rds-data:DeleteItems",
     "rds-data:ExecuteSql",
     "rds-data:ExecuteStatement",
     "rds-data:GetItems",
     "rds-data:InsertItems",
      "rds-data:UpdateItems",
    ],
  resourceArns: [ clusterArn,  `${clusterArn}:*` ],
  scope: this,
});

```

File: packages/@aws-cdk/aws-appsync/README.md
Comment (MrArnoldPalmer) [reply packages/@aws-cdk/aws-appsync/README.md:106]: Yeah we have been piling on every new data source example into this example but starting to separate sections out probably makes sense.

**Comment by BryanPan342** [comment]
revisit and change policy generation for rds cluster to a grant when #9378 is addressed

File: packages/@aws-cdk/aws-appsync/lib/data-source.ts
Comment (kochie) [reply packages/@aws-cdk/aws-appsync/lib/data-source.ts:282]: It's in the [AppSync Doc](https://docs.aws.amazon.com/appsync/latest/APIReference/API_RelationalDatabaseDataSourceConfig.html)

Comment (BryanPan342) [reply packages/@aws-cdk/aws-appsync/lib/data-source.ts:282]: huh i was literally on this page.. i guess i must have glossed over it in my review

**Comment by BryanPan342** [comment]
@kochie there have been some big changes to the code base and I had the bandwidth to implement my suggestions! Please make changes if anything looks out of place

**Comment by MrArnoldPalmer** [comment]
Sorry for the delay here. With conflicts resolved I think I'm good to approved this. @kochie if you're able to resolve those that would be great. If not I'll try to get to it in the next few days.

**Comment by kochie** [comment]
Excellent @MrArnoldPalmer, I'll give it a tidy up and get it done.

**Comment by kochie** [comment]
Hey @MrArnoldPalmer, do you mind giving it a once over again, I had to change some of the objects in tests to get a clean build.

File: packages/@aws-cdk/aws-appsync/README.md
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:150]:
```suggestion
```

dont think this fits here

File: packages/@aws-cdk/aws-appsync/README.md
Comment (BryanPan342) [inline packages/@aws-cdk/aws-appsync/README.md:727]:
```suggestion
```

I believe this is left over from another merge

**Review by BryanPan342 — COMMENTED** [review]
@MrArnoldPalmer @shivlaks LGTM after these changes that I think are a product of merging

**Review by BryanPan342 — APPROVED** [review]
LGTM 🥳

@MrArnoldPalmer passes my check

**Review by MrArnoldPalmer — APPROVED** [review]

**Pull request merged**
